cobigen-cli/cli-systemtest/src/test/java/com/devonfw/cobigen/cli/systemtest/AbstractCliTest.java

```
@@ -32,23 +35,78 @@
 
     devTemplatesPath = new File(AbstractCliTest.class.getProtectionDomain().getCodeSource().getLocation().toURI())
+        .getParentFile().getParentFile().getParentFile().getParentFile().toPath().resolve("cobigen-templates");
+
+    Path utilsPom = new File(AbstractCliTest.class.getProtectionDomain().getCodeSource().getLocation().toURI())
         .getParentFile().getParentFile().getParentFile().getParentFile().toPath().resolve("cobigen-templates")
-        .resolve("templates-devon4j");
+        .resolve("templates-devon4j-tests/src/test/resources/utils/pom.xml");
+
+    // create a temporary directory cobigen-templates/template-sets/adapted containing the template sets
+    Path tempFolderPath = tempFolderTemplates.getRoot().toPath();
+    Path cobigenTemplatePath = tempFolderPath.resolve("cobigen-templates");
+    if (!Files.exists(cobigenTemplatePath)) {
+      Files.createDirectory(cobigenTemplatePath);
+
+      devTemplatesPathTemp = cobigenTemplatePath.resolve(ConfigurationConstants.TEMPLATE_SETS_FOLDER);
+      Path templateSetsAdaptedFolder = devTemplatesPathTemp.resolve(ConfigurationConstants.ADAPTED_FOLDER);
+      Files.createDirectory(devTemplatesPathTemp);
+      Files.createDirectory(templateSetsAdaptedFolder);
+
+      FileUtils.copyDirectory(devTemplatesPath.toFile(), templateSetsAdaptedFolder.toFile());
+
+      try (Stream<Path> files = Files.list(templateSetsAdaptedFolder)) {
+        files.forEach(path -> {
+          if (Files.isDirectory(path)) {
+            Path resourcesFolder = path.resolve("src/main/resources");
+            Path templatesFolder = path.resolve(ConfigurationConstants.TEMPLATE_RESOURCE_FOLDER);
+            if (Files.exists(resourcesFolder) && !Files.exists(templatesFolder)) {
+              try {
+                Files.move(resourcesFolder, templatesFolder);
+              } catch (IOException e) {
+                e.printStackTrace();
```

Please throw an exception with a message.